### PR TITLE
Support custom delimiter KeyPerFile configuration source

### LINKED ISSUE #33589: Request for `--` configuration key delimiter support in KeyPerFile
### Is your feature request related to a problem? Please describe.

I'm using secrets stored in Azure Key Vault in a Kubernetes environment. Key Vault does not support the use of `_` characters in a secret name, so we're advised ([here](https://docs.microsoft.com/aspnet/core/security/key-vault-configuration) and [here](https://docs.microsoft.com/aspnet/core/fundamentals/configuration/?view=aspnetcore-5.0#configuration-keys-and-values)) to use a `--` for a configuration key delimiter in Key Vault. Using the Secrets Store CSI driver and provider for Azure Key Vault (per the [documentation](https://docs.microsoft.com/azure/key-vault/general/key-vault-integrate-kubernetes)), secrets are downloaded to key files in a directory. Loading those secrets using the `KeyPerFile` configuration provider would be a great fit, but it only supports `__` as a configuration key delimiter.

### Describe the solution you'd like

Enabling the `KeyPerFile` configuration provider to support a `--` configuration key delimiter would be great. It would also allow me to maintain compatibility in my scenario with the [Azure Key Vault configuration provider](https://docs.microsoft.com/aspnet/core/security/key-vault-configuration).

### Additional context

None

### PR DESCRIPTION
**Pull request by Kahbazi**
Fixes https://github.com/dotnet/aspnetcore/issues/33589

### CODE REVIEW

File: src/Configuration.KeyPerFile/src/KeyPerFileConfigurationSource.cs [hunk 1]
```diff
@@ -48,6 +48,11 @@ public KeyPerFileConfigurationSource()
         /// </summary>
         public int ReloadDelay { get; set; } = 250;
 
+        /// <summary>
+        /// The delimiter used to separate individual keys in a path. Default is __.
+        /// </summary>
```
Comment (pranavkm) [inline src/Configuration.KeyPerFile/src/KeyPerFileConfigurationSource.cs:53]:
```suggestion
        /// The delimiter used to separate individual keys in a path.
        /// </summary>
        /// <value>Default is <c>__</c>.</value>
```

File: src/Configuration.KeyPerFile/src/KeyPerFileConfigurationSource.cs [hunk 2]
```diff
@@ -48,6 +48,11 @@ public KeyPerFileConfigurationSource()
...
+        public string Delimiter { get; set; } = "__";
```
Comment (pranavkm) [inline src/Configuration.KeyPerFile/src/KeyPerFileConfigurationSource.cs:54]:
```suggestion
        public string SectionDelimiter { get; set; } = "__";
```

**Comment by pranavkm** [comment]
* API triage: Rename `Delimiter` -> `SectionDelimiter`. API approved.

**Review by BrennanConroy — APPROVED** [review]

**Comment by BrennanConroy** [comment]
Thanks @Kahbazi !

**Pull request merged**